source/Mlos.Python/mlos/Optimizers/OptimizerBase.py

```diff
@@ -91,10 +91,23 @@ def optimum(self, optimum_definition: OptimumDefinition = OptimumDefinition.BEST
         if not len(features_df.index):
             raise ValueError("Can't compute optimum before registering any observations.")
 
+        if context is not None and optimum_definition != OptimumDefinition.BEST_PREDICTED_WITHIN_CONTEXT:
+                raise ValueError(f"{optimum_definition} not supported if context is provided.")
+
         if optimum_definition == OptimumDefinition.BEST_OBSERVATION:
             return self._best_observation_optimum(features_df=features_df, objectives_df=objectives_df)
+        elif optimum_definition == OptimumDefinition.BEST_PREDICTED_WITHIN_CONTEXT:
+            if context is None:
+                raise ValueError(f"{optimum_definition} requires context to be not None.")
+            return self._optimum_within_context(context=context)
         return self._prediction_based_optimum(features_df=features_df, optimum_definition=optimum_definition, alpha=alpha)
 
+    @trace()
+    def _optimum_within_context(self, context: pd.DataFrame):
+        greedy_utility = GreedyUtilityFunction(

```

Comment:
This seems a bit strange since now the ``OptimizerBase`` needs to know about ``self.surrogate_model``, but I'm not sure how else to do it (without moving the optimum dispatching into ``BayesianOptimizer``). @byte-sculptor opinions?